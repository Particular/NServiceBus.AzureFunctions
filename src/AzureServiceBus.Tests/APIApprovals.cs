namespace AzureServiceBus.Tests
{
    using NServiceBus.AzureFunctions.AzureServiceBus;
    using NUnit.Framework;
    using Particular.Approvals;
    using PublicApiGenerator;

    [TestFixture]
    public class APIApprovals
    {
        [Test]
        public void Approve()
        {
            var publicApi = ApiGenerator.GeneratePublicApi(typeof(AzureServiceBusTriggerEndpoint).Assembly, excludeAttributes: new[] { "System.Runtime.Versioning.TargetFrameworkAttribute" });
            Approver.Verify(publicApi);
        }
    }
}