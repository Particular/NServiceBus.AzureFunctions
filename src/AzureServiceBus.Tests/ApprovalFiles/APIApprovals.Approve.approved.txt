namespace NServiceBus.AzureFunctions.AzureServiceBus
{
    public class AzureFunctionEndpoint : NServiceBus.Serverless.ServerlessEndpoint<Microsoft.Azure.WebJobs.ExecutionContext>
    {
        public AzureFunctionEndpoint(System.Func<Microsoft.Azure.WebJobs.ExecutionContext, NServiceBus.Serverless.ServerlessEndpointConfiguration> configurationFactory) { }
    }
    public class AzureServiceBusTriggeredEndpointConfiguration : NServiceBus.Serverless.ServerlessEndpointConfiguration
    {
        public AzureServiceBusTriggeredEndpointConfiguration(string endpointName) { }
    }
    public class static AzureServiceBusTriggerExtensions
    {
        public static System.Threading.Tasks.Task Process(this NServiceBus.AzureFunctions.AzureServiceBus.AzureFunctionEndpoint endpoint, Microsoft.Azure.ServiceBus.Message message, Microsoft.Azure.WebJobs.ExecutionContext executionContext) { }
    }
}